### DISCUSSION TITLE
[HW2]  Can I use nn.embedding in glove based models with pretrained weights.?

### DISCUSSION THREAD

**Discussion by sorobedio** [seominjoon/kaist-ai605#127] [Q&A] [answered]
Hello everyone.
is it Okay to use nn.embedding in problem 3.2 with the pretrained weights?
I used it in both glove based cases but my recall is  far behind tf-idf and bm25. 
I wonder if embedding average does mean:
1-I should average the 100d embedding weights  vector for each word.

**Comment by seominjoon** [comment, marked as answer]
tf-idf does pretty well actually in SQuAD, better than dense, so your result is not unexpected.

**Reply by sorobedio** [reply]
Okay i see that.
I still don't clearly understand if i am following the guidelines. so i am still expecting an answer of yes or no about whether i can use nn.embedding with glove or no
1- Do we use nn.embedding layers as i am doing 
2- we do not have to use nn.embedding layer.

**Reply by seominjoon** [reply]
Yes, please use 'nn.Embedding'. That is a preferred way.

**Reply by emjay73** [reply]
Well, I did not use nn.Embedding for this assignment. 
Should I reimplement it? using nn.Embedding?

**Reply by seominjoon** [reply]
@emjay73 No, you don't have to use `nn.Embedding`. It is okay to use it, though.
